dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/MetadataDumper.java

```diff
@@ -130,6 +134,32 @@ public void run(@Nonnull ConnectorArguments arguments) throws Exception {
   private <T> T runTask(
       @Nonnull TaskRunContext context, @Nonnull TaskSetState.Impl state, Task<T> task)
       throws MetadataDumperUsageException {
+    T t = executeTask(context, state, task);
+
+    numberOfCompletedTasks += 1;
+
+    long averageTimePerTaskInMillisecond =
+        (System.currentTimeMillis() - startTimeInMilliseconds) / numberOfCompletedTasks;
+
+    int percentFinished = numberOfCompletedTasks * 100 / totalNumberOfTasks;
+    String progressMessage = percentFinished + "% Completed";
+    PROGRESS_LOG.info(progressMessage);
+    LOG.info(progressMessage);
+
+    int remainingTasks = totalNumberOfTasks - numberOfCompletedTasks;
+    long estimatedTimeInMinutes = averageTimePerTaskInMillisecond * remainingTasks / (60 * 1000);
```

Comment:
Instead of converting to minutes, I would suggest using the `DurationFormatUtils`.

See the example usage:
https://github.com/google/dwh-migration-tools/blob/412897cf34fa90355783fb7f27523948cab32a90/dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/ZonedIntervalIterable.java#L98